dvc/remote/local/__init__.py

```diff
@@ -566,3 +569,96 @@ def unprotect(path_info):
     @staticmethod
     def protect(path_info):
         os.chmod(path_info.path, stat.S_IREAD | stat.S_IRGRP | stat.S_IROTH)
+
+    def append_verififcation_suffix(self, string):
+        return string + "." + self.STATUS_VERIFICATION_DIR_SUFFIX
+
+    def get_verification_dir_info(self, checksum):
+        verification_dir_info = self.checksum_to_path_info(checksum)
+        verification_dir_info.path = self.append_verififcation_suffix(
+            verification_dir_info.path
+        )
+
+        return verification_dir_info
+
+    def create_status_verification_dir(self, checksum, dir_info):
+        verification_dir_info = self.get_verification_dir_info(checksum)
+
+        self.makedirs(verification_dir_info)
```

Comment:
I see you are trying tocreate it on each checkout. Maybe we should check if it exists and is valid first and if it is not then re-create it? Also, what if there are some files that are there but corrupted? Right now it looks like your hardlink() method will just silently skip it.

Reply:
I added check for existing verification dir. As to corrupted files: corruption would have to happen after calculating md5 for files, but before creation of verification dir. Do we wan't to cover this use case?